enhancements/agent-installer/automated-workflow-for-agent-based-installer.md

```diff
@@ -0,0 +1,624 @@
+different installation workflows, and want to avoid inventing a new
+one. Therefore we will reuse existing inputs relevant to on-premises
+deployment, borrowing from the IPI installer and other tools such as
+the zero-touch provisioning (ZTP) system in multi-cluster engine
+(MCE).
+
+Users often do not want to store the credentials for the baseboard
+management controllers (BMC) of their hosts inside of the cluster they
+are deploying. Therefore we cannot rely on them to give us those
+credentials during installation, and we must support tools outside of
+OpenShift managing the hosts' boot process. This places more of the
+burden of designing the end-to-end automated workflow on users or our
+solution architects.
+
+### Risks and Mitigations
+
+The inputs to building an on-premises cluster, especially the per-host
+static network settings, can be complex. Manually creating the inputs
+to the installer provides many opportunities to introduce errors. For
+the automated use case, we assume that users are likely to mitigate
+this risk by also automating production of the input files using
+templates or file generators.
+
+We are investigating the options for including basic static validation
+of the network settings in the installer. Doing so requires access
+golang bindings to a Rust library for nmstate, which may make it
+problematic to distribute a single build artifact. Handling the build
+complications and actual validation is left to a future enhancement.
+
+Other mitigation approaches, such as a GUI for constructing the files,
+are left to future enhancements.
+
+The NMStateConfig API is part of the ZTP layer, and could be changed
+independently of the installer, potentially breaking compatibility
+between the 2 tools. The risk of this happening is low, because the
+API is a relatively simple wrapper for nmstate config content that is
+not parsed directly by ZTP. If the API is changed, the new version
+will need to be backwards-compatible for ZTP users, which should allow
+us to bring the installer up to date without significant impact.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. What do we do with the installation logs?
+
+   - Write them to disk on each host in a way that allows someone
+     logging in to find them or allows a Pod to mount the directory
+     they are in and `cat` them to be collected via the OpenShift API.
+   - Expect the orchestration tool to download them before allowing
+     deployment to continue.
+   - Can we take advantage of or replicate the installer's [bootstrap
+     service
+     records](https://github.com/openshift/installer/blob/6d778f911e79afad8ba2ff4301eda5b5cf4d8e9e/docs/dev/bootstrap_services.md)
+     behavior?
+
+2. Is the workflow different for single-node? Do we need to start the
+   service and go through the orchestration steps for one node? If we
+   don't, how do we handle the validations?
+3. How do we implement a feature flag in the installer?
```

Comment:
I think this one is answered - we don't, just build from a separate branch.

Reply:
I removed this question in eeee5402